usaspending_api/search/v2/views/search.py

```diff
@@ -403,3 +406,105 @@ def post(self, request):
+
+        # build sql query filters
+        queryset = award_filter(filters)
+
+        # define what values are needed in the sql query
+        # queryset = queryset.values('action_date', 'federal_action_obligation')
+
+        # build response
+        response = {'limit': limit, 'page': page, 'results': []}
+        results = []
+
+        for award in queryset:
+            row = {}
+            if set(filters["award_type_codes"]) < set(contract_type_mapping):
+                for field in fields:
+                    try:
+                        award_prop = award
+                        for prop in award_contracts_mapping[field].split("__"):
+                            award_prop = getattr(award_prop, prop)
+                    except:
+                        award_prop = None
+                    row[field] = award_prop
+            elif set(filters["award_type_codes"]) < set(loan_type_mapping):  # loans
+                for field in fields:
+                    try:
+                        award_prop = award
+                        for prop in loan_award_mapping[field].split("__"):
+                            award_prop = getattr(award_prop, prop)
+                    except:
+                        award_prop = None
+                    row[field] = award_prop
+            elif set(filters["award_type_codes"]) < set(non_loan_assistance_type_mapping):  # assistance data
+                for field in fields:
+                    try:
+                        award_prop = award
+                        for prop in non_loan_assistance_award_mapping[field].split("__"):
+                            award_prop = getattr(award_prop, prop)
+                    except:
+                        award_prop = None
+                    row[field] = award_prop
+            results.append(row)
+        results = get_pagination(results, limit, page)
+        response["results"] = results
+        return Response(response)
+
+
+class SpendingByAwardCountVisualizationViewSet(APIView):
+
+    def post(self, request):
+        """Return all budget function/subfunction titles matching the provided search text"""
+        json_request = request.data
+        filters = json_request.get('filters', None)
+
+        if filters is None:
+            raise InvalidParameterException('Missing one or more required request parameters: filters')
+
+        # build sql query filters
+        queryset = award_filter(filters)
+
+        # define what values are needed in the sql query
+        # queryset = queryset.values('action_date', 'federal_action_obligation')
+
+        # build response
+        response = {'results': {}}
+        results = {"contracts": 0, "grants": 0, "direct_payments": 0, "loans": 0, "other": 0}
+
+        for award in queryset:
+            if hasattr(award, "latest_transaction") and hasattr(award.latest_transaction, "contract_data") and \
```

Comment:
an `Award` object should always have the attribute `latest_transaction` so wouldn't `hasattr(award, "latest_transaction")` always be true?

Reply:
This is no longer applicable based on changes in https://github.com/fedspendingtransparency/usaspending-api/pull/525